pkg/cloudprovider/aws.go

```diff
@@ -193,7 +193,23 @@ func (a *AWS) GetNodeEgressIPConfiguration(node *corev1.Node, cloudPrivateIPConf
 	if err != nil {
 		return nil, err
 	}
-	networkInterface := networkInterfaces[0]
+	// Locate the first network interface with a valid interface and subnet for given node
+	var networkInterface *ec2.InstanceNetworkInterface
```

Comment:
Prior to your changes we only considered the first interface, we do something very similar on other clouds where we do not allow (yet) configuring egress IPs on secondary interfaces.
What is the reason for considering all interfaces?

Reply:
Good question. My reasoning was along the lines of making the code a bit mode future proof.
If the implementation is best being strict in only expecting the first interface to be supported,
we can do it that way for sure. Let me rewrite it to behave that way.